Refactor syntax unicode mblsound

livecodefraser opened the pull request.

Commit:
Update MCSystemSound{Player,Channel} to use StringRef names

The class MCSoundFinishedOnChannelEvent and the function MCSoundPostSoundFinishedOnChannelMessage have also been updated.

Both Android and iOS have been changed.

runrevmark commented on engine/src/mblandroidsound.cpp:
```diff
@@ -218,11 +218,15 @@ JNIEXPORT void JNICALL Java_com_runrev_android_SoundModule_doSoundFinishedOnChan
+	
+	MCAutoStringRef t_channel_ref;
+	MCAutoStringRef t_sound_ref;
+	/* UNCHECKED */ MCStringCreateWithBytes(t_channel, env->GetStringUTFLength(channel), kMCStringEncodingUTF8, false, &t_channel_ref);
```
There is an MCJavaStringToStringRef method floating around in the ether which you can use here :)

runrevmark commented on engine/src/mbliphonesound.mm:
```diff
@@ -703,25 +705,24 @@ bool MCSystemNextSoundOnChannel(MCStringRef p_channel, MCStringRef& r_sound)
 	for(MCSystemSoundChannel *t_channel = s_sound_channels; t_channel != nil; t_channel = t_channel -> next)
     {
-		if (t_channel -> name != nil)
+		if (!MCStringIsEmpty(t_channel->name))
```
My take here is that t_channel -> name can't be nil, so having an empty name is fine - thus this check is probably unnecessary.

Commit:
Minor fixes

Tidy up a JNI function by using MCJavaStringToStringRef and remove an un-neccessary check for empty names.
